packages/datagrid/src/basickeyhandler.ts

```diff
@@ -57,6 +66,40 @@ class BasicKeyHandler implements DataGrid.IKeyHandler {
    * This will not be called if the mouse button is pressed.
    */
   onKeyDown(grid: DataGrid, event: KeyboardEvent): void {
+    if (grid.editable && 
+      grid.selectionModel!.cursorRow !== -1 &&
+      grid.selectionModel!.cursorColumn !== -1) {
+      const input = String.fromCharCode(event.keyCode);
+      if (/[a-zA-Z0-9-_ ]/.test(input)) {
+        const row = grid.selectionModel!.cursorRow;
+        const column = grid.selectionModel!.cursorColumn;
+        const cell: CellEditor.CellConfig = {
+          grid: grid,
+          row: row,
+          column: column
+        };
+        grid.editorController!.edit(cell, {
+          onCommit: (response: ICellEditResponse) => {
```

Comment:
This logic is the same as the one used in the mousehandler (but slightly different).  Perhaps the commit/cancel behavior should have a default implementation in `editorController` and allow it to be overridden?

Reply:
@blink1073 agreed. I will update.